garglk/wintext.c

```diff
@@ -539,9 +539,12 @@ void win_textbuffer_redraw(window_t *win)
                     dwin->copypos++;
                 }
             }
-            /* add newline to copy buffer */
-            dwin->copybuf[dwin->copypos] = '\n';
-            dwin->copypos++;
+            /* add newline if we reach the end of the line */
+            if(0==ln->len || (rsc+1)==ln->len)
```

Comment:
For consistency with the codebase, I'd ask that this be changed to:

```c
if (ln->len == 0 || ln->len == (rsc+1))
```
The important changes being a space after ``if`` and swapping around the comparisons so the constant is on the right (and so the second comparison is then consistent with the first).  I understand the rationale behind putting the constant on the left-hand side, but the rest of the code doesn't do it this way, and keeping a consistent style is important.  At some point a style guide ought to be made, but until then, using the existing code as the de facto style is the best we've got.

But in any case I agree this functionality is an improvement.